plugins/node/opentelemetry-instrumentation-mysql2/test/mysql.test.ts

```diff
@@ -1210,13 +1205,160 @@ describe('mysql2', () => {
                   JSON.stringify(res)
                 );
                 done();
-              }
-            );
+              });
+            });
+          });
+        });
+
+        it('should extract data from responseHook - poolCluster', done => {
+          poolCluster.getConnection((err, poolClusterConnection) => {
+            assert.ifError(err);
+            const span = provider.getTracer('default').startSpan('test span');
+            context.with(trace.setSpan(context.active(), span), () => {
+              const sql = 'SELECT 1+1 as solution';
+              poolClusterConnection.query(
+                sql,
+                (err, res: mysqlTypes.RowDataPacket[]) => {
+                  assert.ifError(err);
+                  assert.ok(res);
+                  assert.strictEqual(res[0].solution, 2);
+                  const spans = memoryExporter.getFinishedSpans();
+                  assert.strictEqual(spans.length, 1);
+                  assertSpan(spans[0], sql);
+                  assert.strictEqual(
+                    spans[0].attributes[queryResultAttribute],
+                    JSON.stringify(res)
+                  );
+                  done();
+                }
+              );
+            });
           });
         });
       });
     });
   });
+  describe('promise API', () => {
+    let instrumentation: MySQL2Instrumentation;
+
+    let contextManager: AsyncHooksContextManager;
+    const provider = new BasicTracerProvider();
+    const memoryExporter = new InMemorySpanExporter();
+
+    let connection: mysqlTypesProm.Connection;
+    let rootConnection: mysqlTypesProm.Connection;
+    let mysqlTypesPromReload: typeof mysqlTypesProm;
+
+    before(async function () {
+      // cleanup cache for mysql2
+      delete require.cache[require.resolve('../src')];
+      Object.keys(require.cache).forEach(key => {
+        if (key.includes('mysql2/')) {
+          delete require.cache[key];
+        }
+      });
+
+      const { MySQL2Instrumentation } = await import('../src');
+      instrumentation = new MySQL2Instrumentation();
+      instrumentation.enable();
+      instrumentation.disable();
+
+      mysqlTypesPromReload = await import('mysql2/promise');
+
+      if (!shouldTest) {
+        // this.skip() workaround
+        // https://github.com/mochajs/mocha/issues/2683#issuecomment-375629901
+        this.test!.parent!.pending = true;
+        this.skip();
+      }
+      provider.addSpanProcessor(new SimpleSpanProcessor(memoryExporter));
```

Comment:
`addSpanProcessor` is deprecated. We'll soon publish 2.0 of `@opentelemetry/sdk-trace-base` where this will be removed, so it's best to avoid it wherever we can to make our lives easier later on.

The change is only required for this new test code here, you can leave the old one as-is. #2659 will take care of the old ones.

Ref #2645